### PR TITLE
🐞 Clean up and fix inventory sprite handling

### PR DESCRIPTION
**Pull request by glebm**
@thebigMuh Can you please test this or post a save I can test with?

### CODE REVIEW

File: Source/cursor.cpp [hunk 1]
```diff
@@ -133,19 +140,34 @@ void FreeCursor()
...
 }
 
+const CelSprite &GetInvItemSprite(int i)
+{
+	return i < InvItems1Size ? *pCursCels : *pCursCels2;
+}
+
+int GetInvItemFrame(int i)
+{
+	return i < InvItems1Size ? i : i - InvItems1Size;
```
Comment (thebigMuh) [inline Source/cursor.cpp:150]: This _used_ to be an adjustment of -= 179. Which would actually skip the first entry in the hellfire array portion?

Comment (AJenbo) [reply Source/cursor.cpp:150]: Save game for testing https://github.com/diasurgical/devilutionX/files/5373942/hsv.zip

Comment (glebm) [reply Source/cursor.cpp]: Changed and added a leading zero to the array.
The weird thing is that the frame number seems to start at 1 instead of 0

Comment (glebm) [reply Source/cursor.cpp]: @AJenbo Thanks for the test file, seems to work :+1:

**Review by thebigMuh — APPROVED** [review]

File: Source/cursor.cpp [hunk 2]
```diff
@@ -133,19 +142,34 @@ void FreeCursor()
...
 	icursW28 = icursW / 28;
 	icursH28 = icursH / 28;
 }
 
 void NewCursor(int i)
 {
 	pcurs = i;
-	cursW = InvItemWidth[i];
-	cursH = InvItemHeight[i];
+	std::tie(cursW, cursH) = GetInvItemSize(i);
```
Comment (AJenbo) [inline Source/cursor.cpp:172]: We should probably consider making some sort of structure for the cursor, there is also cursx/y afaik.

**Review by AJenbo — APPROVED** [review]

**Pull request merged**